src/engine/tile/tileset_util.cpp

```diff
@@ -13,18 +13,22 @@ TilesetSetBitMaskTile(Tileset &tileset, int mouse_x, int mouse_y)
     };
     int row_tile_clicked = mouse_y / tileset.sprite_sheet.cell_height();
     int col_tile_clicked = mouse_x / tileset.sprite_sheet.cell_width();
-
+    
     if (row_tile_clicked < tileset.sprite_sheet.n_rows && col_tile_clicked < tileset.sprite_sheet.n_cols)
     {
-        int tile_location_x = col_tile_clicked * tileset.sprite_sheet.cell_height();
-        int tile_location_y = row_tile_clicked * tileset.sprite_sheet.cell_width();
+        int tile_location_x = col_tile_clicked * tileset.sprite_sheet.cell_width();
+        int tile_location_y = row_tile_clicked * tileset.sprite_sheet.cell_height();
+
+        int bit_cell_size_height = tileset.sprite_sheet.cell_height() / 3;
+        int bit_cell_size_width = tileset.sprite_sheet.cell_width() / 3;
 
-        int bit_cell_size = tileset.sprite_sheet.cell_height() / 3;
         int relative_mouse_x = mouse_x - tile_location_x;
         int relative_mouse_y = mouse_y - tile_location_y;
 
-        int bit_clicked_row = relative_mouse_y / bit_cell_size;
-        int bit_clicked_col = relative_mouse_x / bit_cell_size;
+        int bit_clicked_row = relative_mouse_y / bit_cell_size_height;
+        int bit_clicked_col = relative_mouse_x / bit_cell_size_width;
+
+
         bit_clicked_row = bit_clicked_row > 2 ? 2 : bit_clicked_row;
```

Comment:
put `2` in a variable and name it what it means

Reply:
maybe... mae a variable like
`MAX_ROWS_WHATEVER = 3`
then do
`bit_clicked_row = bit_clicked_row >= MAX_ROWS_WHATEVER ? MAX_ROWS_WHATEVER - 1 : bit_clicked_row`